src/blockdb/wrapper.cpp

```diff
@@ -0,0 +1,545 @@
+         bestIndexLev.nBits,
+         bestIndexLev.nNonce,
+         bestIndexLev.nStatus,
+         bestIndexLev.nTx
+         );
+
+    LOGA("check5\n");
+    // if the best height of the storage type we are using is higher than any other type, return false
+    if(BLOCK_DB_MODE == SEQUENTIAL_BLOCK_FILES && bestIndexSeq.nHeight > bestIndexLev.nHeight)
+    {
+        LOGA("false3\n");
+        return false;
+    }
+    LOGA("check6\n");
+    if(BLOCK_DB_MODE == DB_BLOCK_STORAGE && bestIndexLev.nHeight > bestIndexSeq.nHeight)
+    {
+        LOGA("false4\n");
+        return false;
+    }
+    LOGA("check7\n");
+    return true;
+}
+
+// use this sparingly, this function will be very disk intensive
+void SyncStorage(const CChainParams &chainparams)
+{
+    if(BLOCK_DB_MODE == SEQUENTIAL_BLOCK_FILES)
+    {
+        CValidationState state; // not used, just added for compatability
+        // get the best block and go to it. we can sync the blocks in reverse to save time once we hit our best
+        uint256 bestHashSeq = pcoinsdbview->GetBestBlockSeq();
+        uint256 bestHashDb = pcoinsdbview->GetBestBlockDb();
+
+        BlockMap::iterator iter;
+        iter = mapBlockIndex.find(bestHashDb);
+        if(iter == mapBlockIndex.end())
+        {
+            LOGA("couldnt find best hash in mapblockindex when attempting to sync storage types \n");
+            return;
+        }
+        /** we dont have the same problem with finding best block in db mode because of the way it is written */
+        CBlockIndex* pindex = iter->second;
+        LOGA("best seq block we before syncing is %s \n", bestHashSeq.GetHex().c_str());
+        LOGA("best db block we have is %s \n", bestHashDb.GetHex().c_str());
+
+        //we should be good to use this data structure, by block 3M it should only consume about 96MB of memory
+        // we use a vector because order is guarenteed and sequential files rely on that
+        std::vector<uint256> hashesToSync;
+        while(pindex->GetBlockHash() != bestHashSeq)
+        {
+            // we will have what we want in reverse order
+            hashesToSync.emplace_back(pindex->GetBlockHash());
+        }
+
+        boost::scoped_ptr<CDBIterator> pcursor(pblockdb->NewIterator());
+        pcursor->Seek(uint256());
+        // Load mapBlockIndex
+        std::vector<uint256>::reverse_iterator hashiter;
+        for (hashiter = hashesToSync.rbegin(); hashiter != hashesToSync.rend(); ++hashiter)
+        {
+            BlockDBValue blockValue;
+            pblockdb->Read(*hashiter, blockValue);
+            CBlock block_lev = blockValue.block;
+            uint256 checkBlockHash = block_lev.GetHash();
+            BlockMap::iterator it;
+            it = mapBlockIndex.find(checkBlockHash);
+            // we should never have more blocks than headers
+            if(it == mapBlockIndex.end())
+            {
+                LOGA("something is very wrong somewhere \n");
+                assert(false);
```

Comment:
why no using `AbortNode` rather than asserting to false?

Reply:
whenever this failure occurred it would assert false later on an expression check anyway. this worked out to be a case of assert false now or wait to do it later. I chose to add one to just do it now.

Reply:
but didn't `AbortNode` just shutdown the node immediatly?